bundles/org.openhab.binding.vektiva/src/main/java/org/openhab/binding/vektiva/handler/VektivaSmarwiHandler.java

```diff
@@ -0,0 +1,254 @@
+            } else {
+                updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                        "got response code: " + resp.getStatus());
+            }
+            // reconnect web socket if not connected
+            if (config.useWebSockets && (session == null || !session.isOpen())
+                    && ThingStatus.ONLINE.equals(getThing().getStatus())) {
+                logger.debug("Initializing WebSocket session");
+                initializeWebSocketSession();
+            }
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, "exception during status getting");
+            logger.debug("Exception during status update", e);
+            session = null;
+        }
+    }
+
+    public synchronized void processStatusResponse(String content) {
+        if (!ThingStatus.ONLINE.equals(getThing().getStatus())) {
+            updateStatus(ThingStatus.ONLINE);
+        }
+        String[] values = content.split("\n");
+
+        updateProperty("type", getPropertyValue(values, "t"));
+        updateProperty(Thing.PROPERTY_FIRMWARE_VERSION, getPropertyValue(values, "fw"));
+        updateProperty("rssi", getPropertyValue(values, "rssi"));
+        updateProperty("name", getPropertyValue(values, "cid"));
+        updateProperty("status", getPropertyValue(values, "s"));
+        updateProperty("error", getPropertyValue(values, "e"));
+        updateProperty("ok", getPropertyValue(values, "ok"));
+        updateProperty("ro", getPropertyValue(values, "ro"));
+        updateProperty("fix", getPropertyValue(values, "fix"));
+
+        if ("1".equals(getPropertyValue(values, "ro"))) {
+            updateStatus(ThingStatus.ONLINE, ThingStatusDetail.NONE, "Online but not ready!");
+        }
+
+        int position = getPropertyValue(values, "pos").equals("o") ? 0 : 100;
+        if (position == 0 && lastPosition != -1) {
+            position = lastPosition;
+        }
+        updateState(CHANNEL_CONTROL, new PercentType(position));
+    }
+
+    private String getPropertyValue(String[] values, String property) {
+        for (String val : values) {
+            String[] keyVal = val.split(":");
+            if (keyVal.length != 2) {
+                continue;
+            }
+            String key = keyVal[0];
+            String value = keyVal[1];
+            if (property.equals(key)) {
+                return value;
+            }
+        }
+        return "N/A";
+    }
+
+    private @Nullable Session createSession() {
+        String url = "ws://" + config.ip + "/ws";
+        URI uri = URI.create(url);
+
+        try {
+            // The socket that receives events
+            VektivaSmarwiSocket socket = new VektivaSmarwiSocket(this);
+            // Attempt Connect
+            Future<Session> fut = webSocketClient.connect(socket, uri);
+            // Wait for Connect
+            return fut.get();
+        } catch (Exception ex) {
```

Comment:
Same question about catching `Exception`. Can this be specific exceptions and `InterruptException` handling?

Reply:
fixed